### ISSUE TITLE
Failing to deploy on GCE: "bad metadata syntax"

### ISSUE THREAD

**Issue by johnsercel** [jlund/streisand#553]
On GCE, gets through configuring the network, setting firewall rules, gets default SSH key, but then fails at:

TASK [genesis-google : Create the GCE instance] ********************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "bad metadata syntax"}

Tried multiple regions, same. 

Also tried giving the service account both Compute Engine Admin (V1) and (beta) roles, and then only one or the other. But all combinations the same.

**Comment by asantamaria** [comment]
Experienced same issue after trying multiple times although I haven't tried other regions besides the GCE default as @johnsercel  has. Below is the session ending in "bad metadata syntax" error.

Worth noting that I did successfully get streisand installed and running though deleted VM on GCE so trying again.

```
PLAY [Provision the GCE Server] ************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [Set the Google Compute Engine Zone fact] *********************************
ok: [localhost]

TASK [Register JSON file contents] *********************************************
changed: [localhost]

TASK [Set JSON file contents fact] *********************************************
ok: [localhost]

TASK [Set the Google Compute Engine Service Account Email] *********************
ok: [localhost]

TASK [Set the Google Compute Engine Project ID] ********************************
ok: [localhost]

TASK [gce-network : Create network] ********************************************
ok: [localhost]

TASK [gce-network : Open the SSH port in the GCE firewall] *********************
ok: [localhost]

TASK [gce-network : Open necessary TCP ports across most services in the GCE firewall] ***
ok: [localhost]

TASK [gce-network : Open the L2TP/IPsec ports in the GCE firewall] *************
ok: [localhost]

TASK [gce-network : Open the OpenConnect (ocserv) port in the GCE firewall] ****
ok: [localhost]

TASK [gce-network : Open the OpenVPN port in the GCE firewall] *****************
ok: [localhost]

TASK [gce-network : Open the Shadowsocks port in the GCE firewall] *************
ok: [localhost]

TASK [gce-network : Open the WireGuard port in the GCE firewall] ***************
ok: [localhost]

TASK [genesis-google : Get the default SSH key] ********************************
changed: [localhost]

TASK [genesis-google : Create the GCE instance] ********************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "bad metadata syntax"}
	to retry, use: --limit @/Users/asantamaria/streisand/playbooks/google.retry

PLAY RECAP *********************************************************************
localhost                  : ok=15   changed=2    unreachable=0    failed=1   
```

**Comment by asantamaria** [comment]
Was successful in installing streisand on Digital Ocean. Thereafter installed streisand successfully on GCE. For reference, generated a "Computer Engine default service account" service account key.

The _revelatory_ step while installing on Digital Ocean was creating the RSA key pair. There needs to be two files; a private (id_rsa) and public key (id_rsa.pub) in ~.ssh (I'm on on OSX). After my initial install of GCE I deleted the VM and started again. I began deleting files without thinking. And I know I only had one file in ~.ssh when I tried to install streisand on GCE again.

The Digital Ocean instructions to generate key pair are [here](https://www.digitalocean.com/community/tutorials/how-to-use-ssh-keys-with-digitalocean-droplets).

**Comment by cpu** [comment]
@CorbanR Do you feel up to looking into this GCE issue as well as https://github.com/jlund/streisand/issues/568 ? It's not immediately clear if the error @johnsercel saw and the SSH key problem described by @asantamaria are related.

@johnsercel Does the provisioning get far enough through that you can try SSHing to the host? Have you double checked the private/public SSH keypair on your host machine are well formed & in the correct place?

**Comment by CorbanR** [comment]
@cpu Sure. Ill admin though that I'm having a hard time recreating the issues.

**Comment by cpu** [comment]
@CorbanR No pressure :-) If you can't repro you can't repro. Thanks for looking!

**Comment by johnsercel** [comment]
After failing on GCE, I did have success on Digital Ocean. I'll try again this week on GCE, taking into account what @asantamaria mentioned, see if I can still reproduce the issue, and report back.

**Comment by CorbanR** [comment]
@asantamaria @johnsercel the GCE playbook does assume that you have GCE JSON credentials generated by following these directions https://docs.ansible.com/ansible/guide_gce.html#credentials. As well as your public public key located at the default location `~/.ssh/id_rsa.pub`

It would be helpful if you could give me a big more info. 

1. OS: OSX, Ubuntu etc. 
2. Ansible version
3. apache-libcloud version

**Comment by asantamaria** [comment]
1. 
```
Paris:~ asantamaria$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.12.4
BuildVersion:	16E195
```
2. 
```
Paris:~ asantamaria$ ansible --version
ansible 2.2.2.0
```
3. 
```
Paris:~ asantamaria$ pip list
apache-libcloud (1.5.0)
```

Yes, I downloaded the GCE JSON credentials according to the directions (I moved the file to ~/streisand dir) and have the public key located in default location. 

```
Paris:~ asantamaria$ ls .ssh
id_rsa		id_rsa.pub	known_hosts
```
The first time I installed streisand I did not experience any problem. However, I deleted the VM on GCE and began naively "uninstalling" locally by deleting files that were created during the streisand setup. Mistake.

Upon attempting a second install is when the "bad metadata syntax" error arose. 

However, I was able to successfully install streisand a second time on GCE but only after I installed on Digital Ocean. While reading the Digital Ocean instructions on generating a key pair I then realized that I was missing the public key in ~.ssh. After installing successfully on Digital Ocean (with public key created) I then tried installing on GCE and successfully installed a second time.

My conclusion is that not having the id_rsa_pub key is what caused the error. I'm concluding this because that is the only difference I noticed.

**Comment by CorbanR** [comment]
@asantamaria thank you for reporting back! Not having a id_rsa_pub  in the correct location (or one with bad data) could def cause "bad metadata syntax". 

@johnsercel thanks for your willingness to continually troubleshoot, once your testing has completed please let me if the issue has been resolved.

**Comment by cpu** [comment]
Since it seems like @asantamaria was able to resolve their problem & @johnsercel hasn't replied I'm going to close this issue. We can reopen this or a new issue if the problem persists and we can reproduce it. Thanks all!
